### ISSUE TITLE
RPC access not working after 1.9.7

### ISSUE THREAD

**Issue by esocode** [paritytech/parity#8622]
> I'm running:
>
> - **Which Parity version?**: 1.10.+
> - **Which operating system?**: Windows 10
> - **How installed?**: via installer
> - **Are you fully synchronized?**: yes
> - **Which network are you connected to?**: ethereum
> - **Did you try to restart the node?**: yes

I have two parity nodes running, configured (via config.toml) to accept remote RPC requests in the local network. Up to version 1.9.7 accessing the RPC interface from another system in the network works without problems. But after installing 1.10.3 RPC access is not working, not even locally on the nodes although the "Node Status" page displays RPC as active on the configured interface. I tried the versions back to 1.10.0 but with none of the new versions does RPC work. I had to revert both nodes to 1.9.7 to get a working RPC interface.

When starting 1.9.5 the Windows firewall popup came up asking to allow Parity to access the network. This didn't happen with any of the newer versions which may be an indication that the network port is not bound at all on starting Parity, although this also didn't happen with 1.9.7 where RPC works.

**Comment by Tbaut** [comment]
Can you describe your configuration (flags, toml file if you have) and the way you call your node?

**Comment by esocode** [comment]
The nodes are started via `ptray.exe ui` without extra parameters and use the config.toml below. This works without problems for 1.9.5 and I could not find any hint in the change logs about changes of the RPC configuration. Any kind of calling fails with a connection error (no response), be it curl, Postman, or my own code.

```yaml
[rpc]
interface = "<the node's IP>" 
hosts = ["all"]
```

**Comment by Tbaut** [comment]
I don't recall any change in RPC either.
To be able to debug it, you should run the node in the [command line](https://wiki.parity.io/FAQ-Basic-Operations,-Configuration,-and-Synchronization.html#how-to-launch-parity-using-the-command-line-on-windows) and read the errors.

**Comment by esocode** [comment]
Just tried that but the log output doesn't seem to indicate a problem (unless _Multi-threaded server is not available on Windows. Falling back to single thread._ is unexpected). 

```log
2018-05-14 15:20:25  Starting Parity/v1.10.3-stable-b9ceda38a-20180507/x86_64-windows-msvc/rustc1.25.0
2018-05-14 15:20:25  Keys path <...>
2018-05-14 15:20:25  DB path <...>
2018-05-14 15:20:25  Path to <...>
2018-05-14 15:20:25  State DB configuration: fast
2018-05-14 15:20:25  Operating mode: passive
2018-05-14 15:20:26  Configured for Foundation using Ethash engine
2018-05-14 15:20:27  Updated conversion rate to Ξ1 = US$697.39 (6828179.5 wei/gas)
2018-05-14 15:20:27  Multi-threaded server is not available on Windows. Falling back to single thread.
2018-05-14 15:20:27  Multi-threaded server is not available on Windows. Falling back to single thread.
2018-05-14 15:20:32  Public node URL: <...>
2018-05-14 15:20:36  Imported #5612393 3eca…7b7b (168 txs, 7.99 Mgas, 653.54 ms, 34.41 KiB) + another 1 block(s) containing 130 tx(s)
...
```

**Comment by Tbaut** [comment]
No error displayed even when doing some RPC calls?
Can you share the calls you're doing (or samples with curl)?

**Comment by esocode** [comment]
The only error message is a timeout, e.g. _curl: (7) Failed to connect to server port 8545: Timed out_. This is just a simple curl request:
```
curl -H "Content-Type: application/json" -X POST --data "{\"jsonrpc\":\"2.0\",\"method\":\"eth_getBlockByNumber\",\"id\":1, \"params\":[\"latest\", true]}" http://<server hostname>:8545
```
And if I use Parity 1.9.5 this call succeeds. As I said, it's as if the port hasn't been opened at all.

**Comment by esocode** [comment]
A small correction: 1.9.7 also works, so it seems to be a problem of the 1.10 versions. I thought I had tried 1.9.7 too but I had only downloaded it.

**Comment by Tbaut** [comment]
There is definitely a problem in your network configuration, firewall, port... You imagine that if RPC stopped working in 1.10, this would have been a major issue.

I can't reproduce your issue and you'd have to give me a detailed overview of your network and your config for me to help.

**Comment by esocode** [comment]
After some more experimenting I discovered that with the 1.10.x version I need to explicitly allow incoming connections for parity.exe in the Windows firewall. With 1.9.x versions this was not necessary, probably because the firewall rules had been created automatically. With the rule present RPC access works again. I had already tried adding such a rule when I encountered the problem but for some reason it didn't work then. My last tries were with the latest 1.10.4 release, maybe that makes a difference. Another reason could be that the Parity installer for Windows directly starts Parity, but with the installer as the parent process and therefore probably with elevated rights. I'm closing this issue as the problem can be alleviated through the Windows firewall.
